docs/query-metadata-style-guide.md

```diff
@@ -0,0 +1,173 @@
+
+## Query tags `@tags`
+
+The `@tags` property is used to define categories that the query relates to. Each query should belong to one (or more, if necessary) of the following four top-level categories:
+
+*   `@tags correctness`–for queries that detect incorrect program behavior.
+*   `@tags maintainability`–for queries that detect patterns that make it harder for developers to make changes to the code.
+*   `@tags readability`–for queries that detect confusing patterns that make it harder for developers to read the code.
+*   `@tags security`–for queries that detect security weaknesses. See below for further information.
+
+There are also more specific `@tags` that can be specified. See, the following pages for more information on the low-level tags:
+
+*   [C/C++ queries](https://wiki.semmle.com/pages/viewpage.action?pageId=19334052)
+*   [C# queries](https://wiki.semmle.com/display/CSHARP/C%23+queries)
+*   [COBOL queries](https://wiki.semmle.com/display/COBOL/COBOL+queries)
+*   [Java queries](https://wiki.semmle.com/display/JAVA/Java+queries)
+*   [JavaScript queries](https://wiki.semmle.com/display/JS/JavaScript+queries)
+*   [Python queries](https://wiki.semmle.com/display/PYTHON/Python+queries)
+
+If necessary, you can also define your own low-level tags to categorize the queries specific to your project or organization. When creating your own tags, you should:
+
+*   Use all lower-case letters, including for acronyms and proper nouns, with no spaces. All characters apart from * and @ are accepted.
+*   Use a forward slash / to indicate a hierarchical relationship between tags if necessary. For example, a query with tag `foo/bar` is also interpreted as also having tag `foo`, but not `bar`.
+*   Use a single-word `@tags` name. Multiple words, separated with hyphens, can be used for clarity if necessary. 
+
+#### Security query `@tags`
+
+If your query is a security query, use one or more `@tags` to associate it with the relevant CWEs. Add `@tags` for the most specific Base Weakness or Class Weakness in [View 1000](https://cwe.mitre.org/data/definitions/1000.html), using the parent/child relationship. For example:
+
+| `@tags security` | `external/cwe/cwe-022`|
+|-|-|
+||`external/cwe/cwe-023` |
+||`external/cwe/cwe-036` |
+||`external/cwe/cwe-073` |
+
+When you tag a query like this, the associated CWE pages from [MITRE.org](http://cwe.mitre.org/index.html) will automatically appear in the reference section of its associated qhelp file. For more information on qhelp files, see [Query help style guide](https://docs.google.com/document/d/14E-bne3sO3boo0jbmD68XMdTZknN5dXSTXBGQeAfj6A/edit#).
+
+## QL area
+
+### Alert  messages
+
+You must define a message in the select clause of an alert query to display with your results. Alert messages are strings that concisely describe the problem that the alert is highlighting and, if possible, also provide some context. For consistency, alert messages should adhere to the following guidelines:
```

Comment:
Suggest changing the first sentence to: "The `select` clause of each alert query defines the alert message that's displayed for each result found by the query."